packages/model-viewer/src/features/annotation.ts

```diff
@@ -0,0 +1,245 @@
+
+/* @license
+ * Copyright 2019 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Vector3} from 'three';
+import {CSS2DObject, CSS2DRenderer} from 'three/examples/jsm/renderers/CSS2DRenderer.js';
+
+import ModelViewerElementBase, {$onResize, $scene, $tick} from '../model-viewer-base.js';
+import {normalizeUnit} from '../styles/conversions.js';
+import {NumberNode, parseExpressions} from '../styles/parsers.js';
+import {Constructor} from '../utilities.js';
+
+const $annotationRenderer = Symbol('annotationRenderer');
+const $updateHotspots = Symbol('updateHotspots');
+const $hotspotMap = Symbol('hotspotMap');
+const $mutationCallback = Symbol('mutationCallback');
+const $observer = Symbol('observer');
+const $addHotspot = Symbol('addHotspot');
+const $removeHotspot = Symbol('removeHotspot');
+
+/**
+ * Hotspots are configured by slot name, and this name must begin with "hotspot"
+ * to be recognized. The position and normal strings are in the form of the
+ * camera-target attribute and default to "0m 0m 0m" and "0m 1m 0m",
+ * respectively.
+ */
+interface HotspotConfiguration {
+  name: string;
+  position?: string;
+  normal?: string;
+}
+
+/**
+ * The Hotspot object is a reference-counted slot. If decrement() returns true,
+ * it should be removed from the tree so it can be garbage-collected.
+ */
+export class Hotspot extends CSS2DObject {
+  public normal: Vector3;
+  private referenceCount: number;
+
+  constructor(config: HotspotConfiguration) {
+    const wrapper = document.createElement('div');
```

Comment:
Let's set `pointer-events: all` on the wrapper element, so that users don't have to opt-in to pointer-events in their hotspots.